### PR TITLE
Support SemanticsFlag for Header

### LINKED ISSUE #15178: Needs support for UIAccessibilityTraitHeader

### PR DESCRIPTION
**Pull request by goderbauer**
Also exposes some previously missed flags in the Semantics widget and CustomPaintSemantics (plus adds tests to never forget new flags).

Fixes https://github.com/flutter/flutter/issues/15178.

### CODE REVIEW

**Comment by tvolkert** [comment]
AppVeyor's not happy

**Comment by goderbauer** [comment]
Nooo! Fixed.

**Review by tvolkert — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/rendering/custom_paint.dart [hunk 1]
```diff
@@ -817,6 +817,21 @@ class RenderCustomPaint extends RenderProxyBox {
     if (properties.button != null) {
       config.isButton = properties.button;
     }
+    if (properties.textField != null) {
+      config.isTextField = properties.textField;
+    }
+    if (properties.focused != null) {
+      config.isFocused = properties.focused;
```
Comment (yjbanov) [inline packages/flutter/lib/src/rendering/custom_paint.dart]: I'm worried that we might be conflating "focused" with "currently being edited", which on touch screens are two different things 😕 But I think short of a big API change, this ship might have sailed.

Comment (goderbauer) [reply packages/flutter/lib/src/rendering/custom_paint.dart]: How are they different on touch screens? We do make a pretty strong difference between input focus (which is what this is) and accessibility focus.

Comment (yjbanov) [reply packages/flutter/lib/src/rendering/custom_paint.dart]: On non-touch screens a text field would enter focused state immediately, while on touch screens it needs to be activated. But yeah, if focus and accessibility focus are separated, then we're good.

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm

**Pull request merged**